include/votca/tools/graph.h

```diff
@@ -0,0 +1,126 @@
+/*
+ *            Copyright 2009-2018 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <iostream>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+#include <votca/tools/edgecontainer.h>
+#include <votca/tools/graphnode.h>
+#include <votca/tools/name.h>
+
+#ifndef _VOTCA_TOOLS_GRAPH_H
+#define _VOTCA_TOOLS_GRAPH_H
+
+namespace votca {
+namespace tools {
+
+/**
+ * \brief A graph object that contains the graph nodes and the edges describing
+ *        the bonds between nodes.
+ *
+ */
+
+class GraphNode;
+
+class Graph : public EdgeContainer {
+ private:
+  /// Parameter description 
+  /// @param int - is the index of the graph nodes / vertex ids
+  /// @param GraphNode - this is the node object at each vertex and contains 
+  /// all the informatino that is relevant to that object
+  std::unordered_map<int, GraphNode> nodes_;
```

Comment:
Actually, you could also think about `multimap`, see here:
https://www.fluentcpp.com/2018/04/10/maps-vectors-multimap/

Reply:
Just a tip :)